cube/CMakeLists.txt

```diff
@@ -318,3 +324,36 @@ if(APPLE)
 else()
     install(TARGETS vkcubepp RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
 endif()
+
+# ----------------------------------------------------------------------------
+# vkcube-wayland
+
+if(UNIX AND NOT APPLE) # i.e. Linux
+    if(BUILD_WSI_WAYLAND_SUPPORT AND EXISTS ${WAYLAND_PROTOCOLS_PATH}/unstable/xdg-decoration/xdg-decoration-unstable-v1.xml)
+        set(CUBE_INCLUDE_DIRS ${WAYLAND_CLIENT_INCLUDE_DIR} ${CUBE_INCLUDE_DIRS})
+        link_libraries(${WAYLAND_CLIENT_LIBRARIES})
+        set(OPTIONAL_WAYLAND_DATA_FILES
+            ${CMAKE_CURRENT_BINARY_DIR}/xdg-shell-code.c
+            ${CMAKE_CURRENT_BINARY_DIR}/xdg-shell-client-header.h
+            ${CMAKE_CURRENT_BINARY_DIR}/xdg-decoration-code.c
+            ${CMAKE_CURRENT_BINARY_DIR}/xdg-decoration-client-header.h)
+        include_directories(${CMAKE_CURRENT_BINARY_DIR})
+        wayland_setup()
+        if(${CMAKE_SYSTEM_PROCESSOR} STREQUAL ${CMAKE_HOST_SYSTEM_PROCESSOR})
+            add_executable(vkcube-wayland
+                           cube.c
+                           ${PROJECT_SOURCE_DIR}/cube/cube.vert
+                           ${PROJECT_SOURCE_DIR}/cube/cube.frag
+                           cube.vert.inc
+                           cube.frag.inc
+                           ${OPTIONAL_WAYLAND_DATA_FILES})
+            target_link_libraries(vkcube-wayland Vulkan::Vulkan)
+            target_compile_definitions(vkcube-wayland PUBLIC VK_USE_PLATFORM_WAYLAND_KHR)
+            include(CheckLibraryExists)
+            CHECK_LIBRARY_EXISTS("rt" clock_gettime "" NEED_RT)
+            if (NEED_RT)
+                target_link_libraries(vkcube-wayland rt)
+            endif()
+        endif()
+    endif()
+endif()
```

Comment:
I wonder if this would be simpler if you created a single library that was the main code but then created 2 executables, one for wayland directly and one that is the existing `vkcube` which will be compiled with the selected windowing system.

This would reduce the code duplication going on here, which I can accept but don't particularly like.